### ISSUE TITLE
Layer vs Component

### ISSUE THREAD

**Issue by ricfio** [phparkitect/arkitect#236]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | no
| RFC         | yes
| BC Break    | yes

#### Summary

I'm not as experienced as you are, so I might say nonsense... why do not use always 'layer' instead of 'component' ?
I have recently read "Clean Architecture" (Robert C. Martin), and now I have some dubts about the best terminology to use:
- Architecture::withComponents() => Architecture::withLayers()
- component(...) => layer(...)
- mayDependOnComponents(...) => mayDependOnLayers(...)
- shouldNotDependOnAnyComponent(...) => shouldNotDependOnAnyLayer(...)
and so on... 

See the snippet in your README.md:
```
    $layeredArchitectureRules = Architecture::withComponents()
        ->component('Controller')->definedBy('App\Controller\*')
        ->component('Service')->definedBy('App\Service\*')
        ->component('Repository')->definedBy('App\Repository\*')
        ->component('Entity')->definedBy('App\Entity\*')

        ->where('Controller')->mayDependOnComponents('Service', 'Entity')
        ->where('Service')->mayDependOnComponents('Repository', 'Entity')
        ->where('Repository')->mayDependOnComponents('Entity')
        ->where('Entity')->shouldNotDependOnAnyComponent()

        ->rules();
```

**Comment by pfazzi** [comment]
Hello @ricfio, thanks for your question. In my first implementation of this feature, I was using the term layer. Later, I added components and modules as aliases of layers, so there were these three options. Since the validation logic of dependencies between modules, layers and components is almost the same, we decided to leave the most generic term. Component seemed to us to be the most generic term,  so we dropped layer and module.
